code/modules/antagonists/eldritch_cult/eldritch_magic.dm

```diff
@@ -0,0 +1,488 @@
+		INVOKE_ASYNC(src, .proc/fire_line, user,T)
+		T = line_target(25, range, X, user)
+		INVOKE_ASYNC(src, .proc/fire_line, user,T)
+	return ..()
+
+/obj/effect/proc_holder/spell/pointed/ash_final/proc/line_target(offset, range, atom/at , atom/user)
+	if(!at)
+		return
+	var/angle = ATAN2(at.x - user.x, at.y - user.y) + offset
+	var/turf/T = get_turf(user)
+	for(var/i in 1 to range)
+		var/turf/check = locate(user.x + cos(angle) * i, user.y + sin(angle) * i, user.z)
+		if(!check)
+			break
+		T = check
+	return (getline(user, T) - get_turf(user))
+
+/obj/effect/proc_holder/spell/pointed/ash_final/proc/fire_line(atom/source, list/turfs)
+	var/list/hit_list = list()
+	for(var/turf/T in turfs)
+		if(istype(T, /turf/closed))
+			break
+
+		for(var/mob/living/L in T.contents)
+			if(L.anti_magic_check())
+				L.visible_message("<span class='danger'>Spell bounces off of [L.real_name]!</span>","<span class='danger'>The spell bounces off of you!</span>")
+				continue
+			if(L in hit_list || L == source)
+				continue
+			hit_list += L
+			L.adjustFireLoss(20)
+			to_chat(L, "<span class='userdanger'>You're hit by [source]'s fire breath!</span>")
+
+		new /obj/effect/hotspot(T)
+		T.hotspot_expose(700,50,1)
+		// deals damage to mechs
+		for(var/obj/mecha/M in T.contents)
+			if(M in hit_list)
+				continue
+			hit_list += M
+			M.take_damage(45, BURN, "melee", 1)
+		sleep(1.5)
+
+/obj/effect/proc_holder/spell/targeted/shapeshift/eldritch
+	invocation = "SH'PE"
+	invocation_type = "whisper"
+	clothes_req = FALSE
+	action_background_icon_state = "bg_ecult"
+	possible_shapes = list(/mob/living/simple_animal/mouse,\
+		/mob/living/simple_animal/pet/dog/corgi,\
+		/mob/living/simple_animal/hostile/carp,\
+		/mob/living/simple_animal/bot/secbot, \
+		/mob/living/simple_animal/pet/fox,\
+		/mob/living/simple_animal/pet/cat )
+
+/obj/effect/proc_holder/spell/targeted/emplosion/eldritch
+	name = "Energetic Pulse"
+	invocation = "E'P"
+	invocation_type = "whisper"
+	clothes_req = FALSE
+	action_background_icon_state = "bg_ecult"
+	range = -1
+	include_user = TRUE
+	charge_max = 300
+	emp_heavy = 6
+	emp_light = 10
+
+/obj/effect/proc_holder/spell/aoe_turf/fire_cascade
+	name = "Fire Cascade"
+	desc = "creates hot turfs around you."
+	school = "transmutation"
+	charge_max = 300 //twice as long as mansus grasp
+	clothes_req = FALSE
+	invocation = "C'SC'DE"
+	invocation_type = "whisper"
+	range = 4
+	action_icon = 'icons/mob/actions/actions_ecult.dmi'
+	action_icon_state = "fire_ring"
+	action_background_icon_state = "bg_ecult"
+
+/obj/effect/proc_holder/spell/aoe_turf/fire_cascade/cast(list/targets, mob/user = usr)
+	INVOKE_ASYNC(src, .proc/fire_cascade, user,range)
+
+/obj/effect/proc_holder/spell/aoe_turf/fire_cascade/proc/fire_cascade(atom/centre,max_range)
+	playsound(get_turf(centre), 'sound/items/welder.ogg', 75, TRUE)
+	var/_range = 1
+	var/prev_range = 0
+	for(var/i = 0, i <= max_range,i++)
+		for(var/turf/T in range(_range,centre)-range(prev_range,centre))
```

Comment:
An direct boundary proc would probably be better here. Since this will get very expensive with higher ranges.